src/commands/run.rs

```diff
@@ -402,20 +462,25 @@ impl RunCommand {
             .context(format!("failed to instantiate {:?}", self.module))?;
 
         // If a function to invoke was given, invoke it.
-        if let Some(name) = self.invoke.as_ref() {
-            self.invoke_export(store, linker, name)
+        let func = if let Some(name) = &self.invoke {
+            self.find_export(store, linker, name)?
         } else {
-            let func = linker.get_default(&mut *store, "")?;
-            self.invoke_func(store, func, None)
-        }
+            linker.get_default(&mut *store, "")?
+        };
+
+        // Finish all lookups before starting any epoch timers.
+        let finish_epoch_handler = self.setup_epoch_handler(store);
+        let result = self.invoke_func(store, func, self.invoke.as_deref());
+        finish_epoch_handler()?;
```

Comment:
I think that the error handling here may want to be a bit more nuanced perhaps. For example if `invoke_func` failed and this finalization also fails, the original failure will be masked by the writing of the profile. 

This concern of mine is additionally shared with the writing of a core dump after a trap, which IIRC will print out an error but won't return the core dump writing error further up the stack. Could the logic there and here be unified perhaps into a sort of "perform a fallible operation post-wasm-execution"?